spec/index.bs

```diff
@@ -884,20 +884,90 @@ The [=subresources=] of textures included in the views provided to
+[=texture subresource=] in texels that includes the possible extra padding
+to form complete [=texel blocks=] in the [=subresource=]. It is calculated by this procedure:
 
-  - For pixel-based {{GPUTextureFormat|GPUTextureFormats}}, the [=physical size=] is always equal to the size of the [=texture subresource=]
-    used in the sampling hardware.
-  - [=Textures=] in block-based compressed {{GPUTextureFormat|GPUTextureFormats}} always have a [=mipmap level=] 0 whose {{GPUTexture/[[descriptor]]}}.{{GPUTextureDescriptor/size}}
-    is a multiple of the [=texel block size=], but the lower mipmap levels might not be
-    multiples of the [=texel block size=] and can have paddings.
+<div algorithm>
+    <dfn abstract-op>Physical texture subresource extent</dfn>(descriptor, mipLevel)
 
-<div class="example">
-Considering a {{GPUTexture}} in BC format whose {{GPUTexture/[[descriptor]]}}.{{GPUTextureDescriptor/size}} is {60, 60, 1}, when sampling
-the {{GPUTexture}} at [=mipmap level=] 2, the sampling hardware uses {15, 15, 1} as the size of the [=texture subresource=],
-while its [=physical size=] is {16, 16, 1} as the block-compression algorithm can only operate on 4x4 [=texel blocks=].
+    **Arguments:**
+        - {{GPUTextureDescriptor}} |descriptor|
+        - {{GPUSize32}} |mipLevel|
+
+    **Returns:** {{GPUExtent3DDict}}
+
+    1. Let |extent| be a new {{GPUExtent3DDict}} object.
+    1. Let |logicalExtent| be [=Logical texture subresource extent=](|descriptor|, |mipLevel|).
+    1. If |descriptor|.{{GPUTextureDescriptor/dimension}} is:
+        <dl class="switch">
+            : {{GPUTextureDimension/"1d"}}
+            ::
+                Set |extent|.{{GPUExtent3DDict/width}} to |logicalExtent|.[=Extent3D/width=] rounded up to the nearest multiple of |descriptor|'s [=texel block width=].
+
+                Set |extent|.{{GPUExtent3DDict/height}} to 1.
+
+                Set |extent|.{{GPUExtent3DDict/depthOrArrayLayers}} to 1.
+
+            : {{GPUTextureDimension/"2d"}}
+            ::
+                Set |extent|.{{GPUExtent3DDict/width}} to |logicalExtent|.[=Extent3D/width=] rounded up to the nearest multiple of |descriptor|'s [=texel block width=].
+
+                Set |extent|.{{GPUExtent3DDict/height}} to |logicalExtent|.[=Extent3D/height=] rounded up to the nearest multiple of |descriptor|'s [=texel block height=].
+
+                Set |extent|.{{GPUExtent3DDict/depthOrArrayLayers}} to 1.
+
+            : {{GPUTextureDimension/"3d"}}
+            ::
+                Set |extent|.{{GPUExtent3DDict/width}} to |logicalExtent|.[=Extent3D/width=] rounded up to the nearest multiple of |descriptor|'s [=texel block width=].
+
+                Set |extent|.{{GPUExtent3DDict/height}} to |logicalExtent|.[=Extent3D/height=] rounded up to the nearest multiple of |descriptor|'s [=texel block height=].
+
+                Set |extent|.{{GPUExtent3DDict/depthOrArrayLayers}} to |logicalExtent|.[=Extent3D/height=].
```

Comment:
```suggestion
                Set |extent|.{{GPUExtent3DDict/depthOrArrayLayers}} to |logicalExtent|.[=Extent3D/depthOrArrayLayers=].
```

But also there are no compressed 1D / 3D texture formats in WebGPU so the physical extent for them could just be a passthrough of the logical extent.